src/Workspaces/Core/Portable/Remote/RemoteHostSessionHelpers.cs

```diff
@@ -16,12 +16,12 @@ namespace Microsoft.CodeAnalysis.Remote
     /// </summary>
     internal sealed class SessionWithSolution : IDisposable
     {
-        private readonly RemoteHostClient.Connection _connection;
+        private readonly OwnedDisposable<RemoteHostClient.Connection> _connection;
         private readonly PinnedRemotableDataScope _scope;
 
-        public static async Task<SessionWithSolution> CreateAsync(RemoteHostClient.Connection connection, Solution solution, CancellationToken cancellationToken)
+        public static async Task<SessionWithSolution> CreateAsync(OwnedDisposable<RemoteHostClient.Connection>.Boxed connection, Solution solution, CancellationToken cancellationToken)
```

Comment:
📝 `OwnedDisposable<Connection>.Boxed` is functionally equivalent to `ref OwnedDisposable<Connection>`, but the latter cannot be passed as an argument to an `async` method.

Reply:
I see. so it is to be used when ref can't be used.